doc/api/buffer.md

```diff
@@ -1405,9 +1405,9 @@ changes:
 -->
 
 * `value` {string|Buffer|Uint8Array|integer} What to search for.
-* `byteOffset` {integer} Where to begin searching in `buf`. If negative, then
-  offset is calculated from the end of `buf`. **Default:**
-  `buf.length - 1`.
+* `byteOffset` {integer} The index of the last byte in `buf` to be considered
+  as the beginning of a match. If negative, then offset is calculated
```

Comment:
```suggestion
* `byteOffset` {integer} The last zero-based index (in bytes) of `buf` to be
  considered as the beginning of a match. If negative, then offset is calculated
```

<br>

My 2¢, but given that `Buffer` is a subclass of `Uint8Array`, we may be able to use some of the language used by MDN if it's more suitable.

> The `Buffer` class is a subclass of JavaScript's [`Uint8Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) class and extends it with methods that cover additional use cases.

Refs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/byteOffset

Reply:
> I find the proposed wording harder to understand, but maybe that's just me. Do others think it is either more precise or easier to read? @nodejs/documentation

I was inspired by the documentation concerning the `lastIndexOf` method of string ([https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf)) but we can also draw inspiration from the documentation of TypedArray ([https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/lastIndexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/lastIndexOf)) as suggested by @DerekNonGeneric